### PR TITLE
Add more docs for summarizing technologies.

### PR DESCRIPTION
**Pull request by rsuderman**
Simple explanation of technologies including contribution guides.

### CODE REVIEW

File: docs/website/docs/technologies/index.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Technologies
+
+IREE at it's core is an integration of multiple technologies including:
```
Comment (ScottTodd) [inline docs/website/docs/technologies/index.md:3]: Where do you imagine the line is for including a technology in this list?

I could see most of the acronyms in the diagram at https://github.com/google/iree#architecture-overview getting an entry here.

(I left another comment about page organization, and if the list grows to > 5 items/categories I'd definitely want them merged instead of all having their own pages)

File: docs/website/docs/technologies/mlir.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# MLIR
+
+[MLIR](https://mlir.llvm.org/) is the compiler framework that IREE is built
+around. Beyond the tooling this includes a set of common Dialects and
+transforms that IREE uses for it's code generation.
+
+Any required changes to MLIR should be upstreamed by following the MLIR
+contribution [guide](https://mlir.llvm.org/getting_started/Contributing/).
+IREE periodically syncs to MLIR's HEAD so there may be a noticeable delay
+between contributing to MLIR and having the feature available at IREE head.
```
Comment (ScottTodd) [inline docs/website/docs/technologies/mlir.md:11]: Some of these details might make more sense to include in one of these places instead of on the website:
* https://github.com/google/iree/blob/main/CONTRIBUTING.md
* https://github.com/google/iree/blob/main/docs/developers/developing_iree/contributor_tips.md
* https://github.com/google/iree/blob/main/docs/developers/developing_iree/repository_management.md (which is outdated)

In particular, I don't think we need to walk people through the "this might take a while" and "IREE HEAD vs MLIR HEAD" steps. (some portion of people reading these docs may not know what git is, and I don't think we need to explain all those specifics here)

File: docs/website/docs/technologies/tosa.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Tensor Operator Set Architecture
+
+[TOSA](https://developer.mlplatform.org/w/tosa/) defines a set of common
+tensor operations to most machine learning
+frameworks. TOSA's design has been focused provide a simple intermediate
+representation for ingesting models from a variety of sources while
+guaranteeing the representation can support generating efficient
+execution on CPUs, GPUs, and custom accelerators. These operations
```
Comment (ScottTodd) [inline docs/website/docs/technologies/tosa.md:8]: long sentence and grammar errors ("has been focused provide")

File: docs/website/docs/technologies/tosa.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+support both floating point and fixed-point (quantized) operations.
+
+IREE uses TOSA as a prioritzed ingestion dialect, transforming multiple
+ML-platform ingestiong formats into a TOSA compatible set of operations.
```
Comment (ScottTodd) [inline docs/website/docs/technologies/tosa.md:12]: a few typos in here

```suggestion
IREE uses TOSA as a prioritized ingestion dialect, transforming multiple
ML-platform ingestion formats into a TOSA compatible set of operations.
```

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: docs/website/mkdocs.yml [hunk 5]
```diff
@@ -118,12 +118,17 @@ nav:
       - TensorFlow Lite: 'bindings/tensorflow-lite.md'
   - 'Extensions':
       - 'extensions/index.md'
```
[the remark this answers is not in the document]
Comment (ScottTodd) [reply docs/website/mkdocs.yml:120]: (Can bikeshed on the naming of "Glossary"/"Technologies"/"Adjacent Projects"/"Dependencies"/etc. based on what you want to focus these pages on.)

File: docs/website/docs/technologies/tosa.md (hunk 3 above)
Comment (rsuderman) [reply docs/website/docs/technologies/tosa.md:8]: Done.

File: docs/website/docs/technologies/mlir.md (hunk 2 above)
Comment (rsuderman) [reply docs/website/docs/technologies/mlir.md:11]: Removed the upstreaming information to prioritize on technologies.

File: docs/website/docs/technologies/index.md (hunk 1 above)
Comment (rsuderman) [reply docs/website/docs/technologies/index.md:3]: Changed to be a glossary for summarizing technology. Probably adding a compilation workflow using the thing Ben brought up in discord.

File: docs/website/docs/technologies/index.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+- [MLIR](./mlir.md)
+- [SPIRV](./spirv.md)
```
[the remark this answers is not in the document]
Comment (rsuderman) [reply docs/website/docs/technologies/index.md:6]: Done.

File: docs/website/docs/technologies/index.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+- [TOSA](./tosa.md)
+
+Users are free to make improvements to any of IREE's core technologies,
+however changes must be upstreamed to their corresponding projects. This
```
[the remark this answers is not in the document]
Comment (rsuderman) [reply docs/website/docs/technologies/index.md:10]: Changed to be a little less negative.

File: docs/website/mkdocs.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (rsuderman) [reply docs/website/mkdocs.yml:120]: Unified everything under a glossary section.

File: docs/website/docs/reference/glossary.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+a simple intermediate representation for ingesting ML models. This simplifies
+model compilation by targetting multiple front-end languages to TOSA's
+intermediate IR which guarantees efficient execution across multiple device
+types.
+
+IREE uses TOSA as a prioritized ingestion dialect, transforming multiple
+ML-platform ingestion formats into a TOSA compatible set of operations.
+Changes to the TOSA specification require submitting a proposal on
+TOSA's platform development
+[page](https://developer.mlplatform.org/w/tosa/#:~:text=Specification%20Contributions)
```
Comment (ScottTodd) [inline docs/website/docs/reference/glossary.md:53]: A few suggestions in here:

* rephrasing some redundant sentences
* "guarantees efficient execution" -> "without compromising on the ability to achieve efficient execution" ;)
* expand the link title for TOSA's "platform development page" from just "page", so it is easier to click

```suggestion
The [TOSA](https://developer.mlplatform.org/w/tosa/) specification defines a
set of common tensor operations to most machine learning frameworks.
This simplifies model compilation as separate front-end frameworks can target
TOSA's intermediate representation without compromising on the ability to
achieve efficient execution across multiple device types.

IREE uses the TOSA MLIR dialect as a prioritized ingestion format, transforming
multiple ML-platform ingestion formats into a TOSA compatible set of operations.

Changes to the TOSA specification require submitting a proposal on TOSA's
[platform development page](https://developer.mlplatform.org/w/tosa/#:~:text=Specification%20Contributions)
```

File: docs/website/docs/reference/glossary.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+## TFLite
+
+[TensorFlow lite](https://www.tensorflow.org/lite) is a model format and
+execution system for performing on device inference. IREE supports TFLite
+flatbuffers translation into [TOSA](#tosa-dialect) operations and compilation
```
Comment (ScottTodd) [inline docs/website/docs/reference/glossary.md:59]: is this `#tosa-dialect` link correct, with the changes to the page structure?

File: docs/website/mkdocs.yml [hunk 10]
```diff
@@ -80,7 +80,11 @@ markdown_extensions:
       options:
         custom_icons:
           - overrides/.icons
-  - pymdownx.superfences
+  - pymdownx.superfences:
+      custom_fences:
+        - name: mermaid
+          class: mermaid
+          format: !!python/name:pymdownx.superfences.fence_code_format
```
Comment (ScottTodd) [inline docs/website/mkdocs.yml:87]: could split this into a separate PR, and https://squidfunk.github.io/mkdocs-material/reference/diagrams/ requires insiders (which we haven't set up yet)

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: docs/website/mkdocs.yml (hunk 10 above)
Comment (rsuderman) [reply docs/website/mkdocs.yml]: Removed.

File: docs/website/docs/reference/glossary.md (hunk 9 above)
Comment (rsuderman) [reply docs/website/docs/reference/glossary.md]: Fixed, should not have been #tosa-dialect

File: docs/website/docs/reference/compilation.md [hunk 11]
```diff
@@ -0,0 +1,5 @@
+
+The map below shows how different front-ends are compiled to separate targets
+via a variety of IR.
+
+![compilation](compilation_diagram.svg)
```
[the remark this answers is not in the document]
Comment (rsuderman) [reply docs/website/docs/reference/compilation.md]: Removed.

**Review by ScottTodd — APPROVED** [review]
